internal/services/automation/automation_variables_data_source.go

```diff
@@ -0,0 +1,289 @@
+package automation
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/automation/2020-01-13-preview/variable"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/automation/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+)
+
+func dataSourceAutomationVariables() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Read: dataSourceAutomationVariablesRead,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Read: pluginsdk.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"resource_group_name": commonschema.ResourceGroupName(),
+
+			"automation_account_name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ValidateFunc: validate.RunbookName(),
+			},
+
+			"bool": {
+				Type:     pluginsdk.TypeList,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					// How do I make this related to the specific type?
```

Comment:
Are these comments relevant or resolved?

Reply:
Kinda both 🙂  I was hoping to try and return all the variables in a single property, and set the type of the value depending on the variable type. But I don't know enough Go to know if it is even possible, so I took the easy route and made them all separate properties.